packages/capabilities/src/store.js

```diff
@@ -64,16 +73,16 @@ export const add = capability({
     origin: Link.optional(),
   }),
   derives: (claim, from) => {
-    const result = equalLink(claim, from)
-    if (result.error) {
-      return result
-    } else if (claim.nb.size !== undefined && from.nb.size !== undefined) {
-      return claim.nb.size > from.nb.size
-        ? fail(`Size constraint violation: ${claim.nb.size} > ${from.nb.size}`)
-        : ok({})
-    } else {
-      return ok({})
-    }
+    return (
+      and(equalWith(claim, from)) ||
+      and(checkLink(claim.nb.link, from.nb.link, 'nb.link')) ||
+      and(checkQuota(claim.nb.size, from.nb.size, 'nb.size')) ||
+      and(
+        checkLink(claim.nb.piece.link, from.nb.piece.link, 'nb.piece.link')
+      ) ||
+      and(equal(claim.nb.piece.size, from.nb.piece.size, 'nb.piece.size')) ||
```

Comment:
I realize this was not added here but `and` does not read well for me - I'm having real trouble understanding what this code means.